### PR TITLE
JavaFX view only

### PR DESCRIPTION
**Pull request by AustinShalit**
I am seeking a review for a JavaFX version of OutlineViewer.  This PR is not complete and that is why I am making it to the JavaFX branch of outline viewer.  This version is only a viewer and does not let you edit the table in any way.  I will be making future PRs that will all you to edit the data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,26 @@
-# Outline Viewer Project
+[![Build Status](https://travis-ci.org/wpilibsuite/OutlineViewer.svg?branch=master)](https://travis-ci.org/wpilibsuite/OutlineViewer) [![codecov](https://codecov.io/gh/wpilibsuite/OutlineViewer/branch/master/graph/badge.svg)](https://codecov.io/gh/wpilibsuite/OutlineViewer)
```
Comment (JLLeitschuh) [inline README.md]: You can put these on different lines.
Easier to update/read in a diff.

They will show up on the same line in any markdown viewer.

Comment (AustinShalit) [reply README.md]: The more you know

**Comment by AustinShalit** [comment]
I totally forgot to create data types for arrays and raw... 😊

So that will not work until I fix it.

File: build.gradle [hunk 2]
```diff
@@ -1,43 +1,142 @@
...
+    println "Running UI Tests Headless"
+    test {
+        jvmArgs = ['-Djava.awt.headless=true', '-Dtestfx.robot=glass', '-Dtestfx.headless=true', '-Dprism.order=sw', '-Dprism.text=t2k']
+    }
 }
 
-dependencies {
-    compile "edu.wpi.first.wpilib.networktables.java:NetworkTables:+:desktop"
-    compile 'uk.gov.nationalarchives.thirdparty.netbeans:org-netbeans-swing-outline:7.2'
+mainClassName = 'edu.wpi.first.OutlineViewer'
```
Comment (nightpool) [inline build.gradle]: this should be outlineviewer.OutlineViewer? I couldn't get it to run otherwise.

**Comment by nightpool** [comment]
Currently it seems pretty broken:
image

The table setup, for comparison:
image

Looks like you have problems with your table algorithm. I would recommend looking at the code in shuffleboard's NetworkTableTree, especially makeBranches.

**Comment by AustinShalit** [comment]
Added a unit test to make sure that does not break in the future.

**Comment by AustinShalit** [comment]
Still broken

**Comment by AustinShalit** [comment]
In favor of #26

**Pull request closed without merging**